napari_animation/animation.py

```diff
@@ -159,7 +169,27 @@ def _frame_generator(self, canvas_only=True):
 
     @property
     def current_key_frame(self):
-        return self.key_frames[self.frame]
+        try:
+            return self.key_frames.index(self.key_frames.selection._current)
+        except ValueError:
+            warnings.warn("No current frame selected !")
+            return None
+
+    @current_key_frame.setter
+    def current_key_frame(self, frame_index):
+        self.key_frames.selection._current = self.key_frames[frame_index]
+
+    @property
+    def active_key_frame(self):
+        try:
+            return self.key_frames.index(self.key_frames.selection.active)
+        except ValueError:
+            warnings.warn("No active frame selected !")
+            return None
+
+    @active_key_frame.setter
+    def active_key_frame(self, frame_index):
+        self.key_frames.selection.active = self.key_frames[frame_index]
```

Comment:
as mentioned in my comment, I might make these return the objects themselves, rather than their indices.  I might also remove the warning and just return `None` (seems like a very reasonable thing just to have no active/current keyframe).  Finally, unless you have an immediate need for the `@setter`, you could punt on exposing that for now... reducing this all to:

```suggestion
    @property
    def current_key_frame(self):
        return self.key_frames.selection._current

    @property
    def active_key_frame(self):
        return self.key_frames.selection.active
```